src/arrayMethodSort.js

```diff
@@ -4,8 +4,34 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  function defaultCompare(a, b) {
+    if (a.toString() < b.toString()) {
+      return -1;
+    }
+
+    if (a.toString() > b.toString()) {
+      return 1;
+    }
+
+    return 0;
+  }
+
+  [].__proto__.sort2 = function (compareFunction = defaultCompare) {
+    if (this.length === 0) {
+      return this;
+    }
+
+    const compare = compareFunction;
+
+    for (let i = 0; i < this.length - 1; i++) {
+      for (let j = 0; j < this.length - 1 - i; j++) {
+        if (compare(this[j], this[j + 1]) > 0) {
+          [this[j + 1], this[j]] = [this[j], this[j + 1]];
+        }
+      }
+    }
+
+    return this;
```

Comment:
The bubble sort algorithm used here is not the most efficient for sorting large arrays. While it's acceptable for small arrays or for educational purposes, for larger arrays or performance-critical applications, consider implementing a more efficient sorting algorithm like quicksort or mergesort.